pwiz_tools/Skyline/FileUI/ExportMethodDlg.cs

```diff
@@ -54,11 +56,13 @@ public sealed partial class ExportMethodDlg : CreateHandleDebugBase, IMultipleVi
 
         private readonly ExportDlgProperties _exportProperties;
 
+        private CancellationTokenSource _cancellationTokenSource = new CancellationTokenSource();
+
         public ExportMethodDlg(SrmDocument document, ExportFileType fileType)
         {
             InitializeComponent();
 
-            _exportProperties = new ExportDlgProperties(this);
+            _exportProperties = new ExportDlgProperties(this, _cancellationTokenSource);

```

Comment:
You should probably pass _cancellationTokenSource.Token in here.

Objects that just need to know whether the job is cancelled should be given a CancellationToken.
The only objects that should see the CancellationTokenSource are the ones that need to be able to cancel the job.